custom_components/resmed_myair/config_flow.py

```
@@ -100,26 +135,21 @@ async def async_step_user(
-                status, device, self._client = await get_device(
-                    self.hass,
-                    self._data[CONF_USER_NAME],
-                    self._data[CONF_PASSWORD],
-                    self._data[CONF_REGION],
-                )
+                status, device = await self._async_login_and_get_device()
                 if device and status == AUTHN_SUCCESS:
                     _LOGGER.debug("[async_step_user] device: %s", redact_dict(device))
-                    if "serialNumber" not in device:
+                    if not device.serial_number:
                         raise ParsingError("Unable to get Serial Number from Device Data")
```

Ruff (0.15.14)

[warning] Abstract `raise` to an inner function

(TRY301)